2009-04-01  Carlos O'Donell  <carlos@codesourcery.com>

	* csu/init.c: Rename to ...
	* csu/csu-init.c: ... here.
	* csu/Makefile: Use csu-init.
	* Version.def: Add GLIBC_2.9.

Index: Versions.def
===================================================================
RCS file: /cvs/glibc/libc/Versions.def,v
retrieving revision 1.74
diff -u -p -r1.74 Versions.def
--- a/Versions.def	3 Dec 2008 04:19:06 -0000	1.74
+++ b/Versions.def	2 Apr 2009 02:36:54 -0000
@@ -88,6 +88,7 @@ libpthread {
   GLIBC_2.3.4
   GLIBC_2.4
   GLIBC_2.6
+  GLIBC_2.9
   GLIBC_PRIVATE
 }
 libresolv {
Index: csu/init.c
===================================================================
RCS file: csu/init.c
diff -N csu/init.c
--- a/csu/init.c	6 Jul 2001 04:54:45 -0000	1.4
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,27 +0,0 @@
-/* Special startup support.
-   Copyright (C) 1997, 1998 Free Software Foundation, Inc.
-   This file is part of the GNU C Library.
-
-   The GNU C Library is free software; you can redistribute it and/or
-   modify it under the terms of the GNU Lesser General Public
-   License as published by the Free Software Foundation; either
-   version 2.1 of the License, or (at your option) any later version.
-
-   The GNU C Library is distributed in the hope that it will be useful,
-   but WITHOUT ANY WARRANTY; without even the implied warranty of
-   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-   Lesser General Public License for more details.
-
-   You should have received a copy of the GNU Lesser General Public
-   License along with the GNU C Library; if not, write to the Free
-   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
-   02111-1307 USA.  */
-
-#if defined USE_IN_LIBIO && defined __GNUC__ && __GNUC__ >= 2
-
-#include <_G_config.h>
-
-/* This records which stdio is linked against in the application. */
-const int _IO_stdin_used = _G_IO_IO_FILE_VERSION;
-
-#endif
Index: csu/csu-init.c
===================================================================
RCS file: csu/csu-init.c
diff -N csu/csu-init.c
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ b/csu/csu-init.c	2 Apr 2009 02:36:55 -0000
@@ -0,0 +1,27 @@
+/* Special startup support.
+   Copyright (C) 1997, 1998 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, write to the Free
+   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+   02111-1307 USA.  */
+
+#if defined USE_IN_LIBIO && defined __GNUC__ && __GNUC__ >= 2
+
+#include <_G_config.h>
+
+/* This records which stdio is linked against in the application. */
+const int _IO_stdin_used = _G_IO_IO_FILE_VERSION;
+
+#endif
Index: csu/Makefile
===================================================================
RCS file: /cvs/glibc/libc/csu/Makefile,v
retrieving revision 1.79
diff -u -p -r1.79 Makefile
--- a/csu/Makefile	1 Mar 2006 10:35:47 -0000	1.79
+++ b/csu/Makefile	2 Apr 2009 02:36:55 -0000
@@ -40,7 +40,7 @@ omit-deps = $(patsubst %.o,%,$(start-ins
 			     S$(start-installed-name))
 install-lib = $(start-installed-name) g$(start-installed-name) $(csu-dummies)
 distribute = initfini.c gmon-start.c start.c defs.awk \
-	     abi-note.S init.c c not-cancel.h
+	     abi-note.S csu-init.c c not-cancel.h
 generated = version-info.h
 before-compile = $(objpfx)version-info.h
 
@@ -125,7 +125,7 @@ $(objpfx)defs.h: $(objpfx)initfini.s
 endif
 
 ifeq (yes,$(elf))
-extra-objs += abi-note.o init.o
+extra-objs += abi-note.o csu-init.o
 asm-CPPFLAGS += -I$(objpfx).
 endif
 
@@ -143,13 +143,13 @@ ifeq (yes,$(elf))
 # We link the ELF startfile along with a SHT_NOTE section indicating
 # the kernel ABI the binaries linked with this library will require.
 $(objpfx)$(start-installed-name): $(objpfx)start.o $(objpfx)abi-note.o \
-				  $(objpfx)init.o
+				  $(objpfx)csu-init.o
 	$(link-relocatable)
 $(objpfx)S$(start-installed-name): $(objpfx)start.os $(objpfx)abi-note.o \
-				  $(objpfx)init.o
+				  $(objpfx)csu-init.o
 	$(link-relocatable)
 $(objpfx)b$(start-installed-name): $(objpfx)start.ob $(objpfx)abi-note.ob \
-				  $(objpfx)init.ob
+				  $(objpfx)csu-init.ob
 	$(link-relocatable)
 else
 # The startfile is installed under different names, so we just call our
